pkg/cmd/server/origin/master.go

```diff
@@ -460,16 +428,59 @@ func (c *MasterConfig) InitializeObjects() {
 	c.ensureOpenShiftSharedResourcesNamespace()
 }
 
-func (c *MasterConfig) InstallProtectedAPI(apiContainer *genericmux.APIContainer) ([]string, error) {
-	// initialize OpenShift API
+func (c *MasterConfig) InstallProtectedAPI(apiserver *genericapiserver.GenericAPIServer) ([]string, error) {
+	apiContainer := apiserver.HandlerContainer
+	messages := []string{}
 	storage := c.GetRestStorage()
 
-	messages := []string{}
+	// install API groups
+	for gv, gvStorage := range storage {
+		// skip pure-legacy groups as API groups
+		if gv == v1.SchemeGroupVersion {
+			continue
+		}
+		if !registered.IsEnabledVersion(gv) {
+			continue
+		}
+
+		apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(gv.Group)
+		versions := []string{
+			securityapiv1.SchemeGroupVersion.Version,
```

Comment:
Can we move this array outside the loop, seems odd we are constructing it over and over again on each loop pass and it doesn't change at any point.